isos/base.sh

```diff
@@ -69,6 +69,13 @@ cp $DIR/base/yum.conf $(rootfs_dir $PKGDIR)/etc/yum/
 
 # install the core packages
 yum_cached -c $cache -u -p $PKGDIR install filesystem coreutils linux-esx --nogpgcheck -y
+
+
+# Issue 3858: find all kernel modules and unpack them and run depmod against that directory
+find $(rootfs_dir $PKGDIR)/lib/modules -name "*.ko.xz" | xargs xz -d
+KERNEL_VERSION=$(basename $(rootfs_dir $PKGDIR)/lib/modules/*)
+chroot $(rootfs_dir $PKGDIR) depmod $KERNEL_VERSION
```

Comment:
Did you try `depmod -b`?
```
  -b basedir, --basedir basedir
           If your modules are not currently in the (normal) directory > /lib/modules/version, but in a staging area, you can specify a basedir which is prepended to the directory name. This basedir is
           stripped from the resulting modules.dep file, so it is ready to be moved into the normal location. Use this option if you are a distribution vendor who needs to pre-generate the meta-data files
           rather than running depmod again later.
```